### PR TITLE
[MM 24831] Convert LocalizedInput of 'email_to_oauth.tsx' to regular input component

### PR DESCRIPTION
**Pull request by js029**
#### Summary
Converted LocalizedInput components in 'email_to_oauth.tsx' to input and added React-Intl AP to accommodate translated strings

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/24831

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]
Approved in a second, looks great 🥳

File: webapp/channels/src/components/claim/components/email_to_oauth.tsx [hunk 1]
```diff
@@ -119,12 +119,12 @@ const EmailToOAuth = (props: Props) => {
...
                 </p>
                 <div className={classNames('form-group', {'has-error': serverError})}>
-                    <LocalizedInput
+                    <input
                         type='password'
                         className='form-control'
                         name='password'
                         ref={passwordInput}
-                        placeholder={placeholderPasswordMessage}
+                        placeholder={formatMessage({id: t('claim.email_to_oauth.pwd'), defaultMessage: 'Password'})}
```
Comment (larkox) [inline webapp/channels/src/components/claim/components/email_to_oauth.tsx:127]: The `t` function should not be needed here anymore.

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just one small change.

File: webapp/channels/src/components/claim/components/email_to_oauth.tsx (hunk 1 above)
Comment (js029) [reply webapp/channels/src/components/claim/components/email_to_oauth.tsx]: Removed

**Review by larkox — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**